providers/base/provider.go

```diff
@@ -112,3 +112,15 @@ func (p *BaseProvider[K, V]) GetData() map[K]providertypes.Result[V] {
 
 	return cpy
 }
+
+// Type returns the type of data handler the provider uses
+func (p *Provider[K, V]) Type() providermetrics.ProviderType {
+	switch {
+	case p.apiCfg.Enabled && p.api != nil:
+		return providermetrics.API
+	case p.wsCfg.Enabled && p.ws != nil:
+		return providermetrics.WebSockets
+	default:
+		return "unknown"
+	}
```

Comment:
this is a small nit but i'd prefer if the provider type (api, websocket, etc). was defined in the provider types i.e. `"github.com/skip-mev/slinky/providers/types"` package instead of `"github.com/skip-mev/slinky/providers/base/metrics"`. think it makes more sense to define all of the provider type variables in a single place.